src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -558,7 +558,7 @@ void ProcessMarkedTypesWithInterfaces ()
 			foreach (var type in typesWithInterfaces) {
 				// Exception, types that have not been flagged as instantiated yet.  These types may not need their interfaces even if the
 				// interface type is marked
-				if (!Annotations.IsFromStaticInterface (type) && !Annotations.IsInstantiated (type) && !Annotations.IsRelevantToVariantCasting (type))
+				if (!Annotations.IsInstantiated (type) && !Annotations.IsRelevantToVariantCasting (type))
```

Comment:
I would expect this to break the functionality. Static interface method types should not be instantiated (hopefully the tests don't do that) and vast majority of them won't be relevant to variant casting either. So I would expect these to hit the 'continue' here and "escape".
Meaning we'd be left with a type which doesn't implement the static interface after trimming. The tests probably don't validate this yet.

Reply:
Isn't this path used for lazy marking of stuff that `IsInterfaceOverrideThatDoesNotNeedMarked` allowed to be lazy? Since we reject static methods being done lazily, we shouldn't need to do anything for static methods here.

Reply:
Maybe I don't understand this correctly. I think the difference is marking methods versus marking types and interface implementations.

`ProcessOverride` (called via `ProcessVirtualMethod`) will handle the methods, but it doesn't really mark types (only indirectly). `ProcessMarkedTypesWithInterfaces` on the other hand deals only with types and more importantly with `InterfaceImplementation`. The effect is probably very subtle and won't be visible in most tests, but for example `MarkInterfaceImplementation` will mark the attributes on the interface impl, which would not happen for static interfaces if I read it correctly.